_posts/2023-07-30-Server-Metadata.md

```diff
@@ -0,0 +1,35 @@
+---
+layout: post
+title: "Adding metadata to Servers"
+lang: "en"
+heading: "Metadata for servers"
+---
+
+To provide information about Servers or clarify policies for Bots, server owners can add metadata to their servers' welcome message. This page gives a rough format about how the metadata could look like.
+
+<!--more-->
+
+You can add metadata to your server for many purposes. Adding contact information like the E-Mail or Website of the Server administrator helps users to contact the administrator if something goes wrong with the server. Also adding policies for bots allows them to judge if they can or cannot connect to the server.
```

Comment:
It only helps if there's a straightforward way for users to find or use this information. Including this information in clear text is arguably much more useful than hiding it as our HTML will do. If I wanted users to easily contact me, providing a visible email or website address seems foolproof, while a hidden metadata entry is... useless at this point, right? So I'm not sure we've thought through how this guidance would help.

Reply:
Tools can extract it - and probably also jamulus explorer could parse it and display it in a new field.

Reply:
The user's discovery is highly diminished vs. clear text. I don't see how we could beat guidance that says, "Hey, put your email address in your welcome message." Putting it in hidden text gains us... what?

Reply:
The _point_ of "metadata" is _not_ to be human-readable.  It's direct at machines and should be in an easily _machine-parseable_ format.

Anyone can put any human-visible text in their welcome message already.

The idea here was that we'd say, "Hey, if you want to tag up your server so it has machine readable data, here's how we suggest you do it."

Then if someone says "Hey, it'd be cool if you tag your server up with <some info I want to parse> using <this suggested tag>" we already have that idea embedded in the documentation and it's not alien to server operators.

It's _definitely_ never going to be something Jamulus itself is interested in.

Reply:
Why not? Require an admin contact email someday. You could announce new releases via email. It'd be grand.

Reply:
I think this goes against the policy of Jamulus being open. Also alerting people via E-Mail should be opt in.

> The point of "metadata" is not to be human-readable. It's directed at machines and should be in an easily machine-parseable format.

Exactly. I think we should drop the talk about admin contact here.
Focus on a generalized metadata format.

Reply:
I've suggested focusing on bot policy, and also suggested brainstorming good candidates for metadata usage. Surely someone could think of more than one usage of this generalized format before you tout it.